packages/orbit-components/src/Toast/hooks/useSwipe.js

```diff
@@ -0,0 +1,82 @@
+// @flow
+import { useCallback, useEffect, useState } from "react";
+
+export default function useSwipeToDismiss(
+  ref: {| current: HTMLElement | null |},
+  onDismiss: () => void,
+  distanceBeforeDismiss: number,
+  direction: "left" | "right",
+): {| swipeOffset: number, swipeOpacity: number |} {
+  const node = ref.current;
+  const [removing, setRemoving] = useState(false);
+  const [pressedPosition, setPressedPosition] = useState(false);
+  const [positionLeft, setPositionLeft] = useState(0);
+  const [opacity, setOpacity] = useState(1);
+
+  const remove = useCallback(() => {
+    setTimeout(() => {
+      onDismiss();
+    }, 300);
+  }, [onDismiss]);
+
+  const onMouseUp = useCallback(() => {
```

Comment:
Consider moving all of these event handlers inside the hooks where you're using them, it wouldn't result in any significant repetition and the code should be easier to follow. This is the common pattern of attaching and removing event handlers in hooks:

```js
useEffect(() => {
  const handler = () => { /* ... */ }
  el.addEventListener('my-event', handler)
  return () => {
    el.removeEventListener('my-event', handler)
  }
}, [/* ... */])
```

Reply:
isn' it already like that? or what repetition do you mean?